src/app/loadout-builder/ModPicker.tsx

```diff
@@ -0,0 +1,291 @@
+                plugSets[item.bucket.hash].add(socket.socketDefinition.reusablePlugSetHash);
+              } else if (socket.socketDefinition.randomizedPlugSetHash) {
+                plugSets[item.bucket.hash].add(socket.socketDefinition.randomizedPlugSetHash);
+              }
+              // TODO: potentially also add inventory-based mods
+            });
+        }
+      }
+
+      // 2. for each unique socket (type?) get a list of unlocked mods
+      const allMods = Object.values(plugSets).flatMap((sets) => {
+        const unlockedPlugs: number[] = [];
+
+        for (const plugSetHash of sets) {
+          const plugSetItems = itemsForPlugSet(profileResponse, plugSetHash);
+          for (const plugSetItem of plugSetItems) {
+            if (plugIsInsertable(plugSetItem)) {
+              unlockedPlugs.push(plugSetItem.plugItemHash);
+            }
+          }
+        }
+
+        return unlockedPlugs
+          .map((i) => defs.InventoryItem.get(i))
+          .filter((item) => isArmor2Mod(item) && item.collectibleHash)
+          .sort((a, b) => sortMods(a, b));
+      });
+
+      return _.uniqBy(allMods, (mod) => mod.hash);
+    }
+  );
+
+  return (state: RootState, props: ProvidedProps): StoreProps => ({
+    isPhonePortrait: state.shell.isPhonePortrait,
+    buckets: state.inventory.buckets!,
+    language: settingsSelector(state).language,
+    mods: unlockedModsSelector(state, props),
+    defs: state.manifest.d2Manifest!
+  });
+}
+
+interface State {
+  query: string;
+  height?: number;
+  lockedArmor2Mods: LockedArmor2ModMap;
+}
+
+/**
+ * A sheet that allows picking a perk.
+ */
+class ModPicker extends React.Component<Props, State> {
+  state: State = {
+    query: '',
+    lockedArmor2Mods: copy(this.props.lockedArmor2Mods)
+  };
+  private itemContainer = React.createRef<HTMLDivElement>();
+  private filterInput = React.createRef<SearchFilterInput>();
+
+  componentDidMount() {
+    if (this.itemContainer.current) {
+      this.setState({ height: this.itemContainer.current.clientHeight });
+    }
+    if (!this.props.isPhonePortrait && this.filterInput.current) {
+      this.filterInput.current.focusFilterInput();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.itemContainer.current && !this.state.height) {
+      this.setState({ height: this.itemContainer.current.clientHeight });
+    }
+  }
+
+  render() {
+    const { defs, mods, language, onClose, isPhonePortrait } = this.props;
+    const { query, height, lockedArmor2Mods } = this.state;
+
+    const Armor2ModPlugCategoriesTitles = {
+      [Armor2ModPlugCategories.general]: 'LB.General',
+      [Armor2ModPlugCategories.helmet]: 'LB.Helmet',
+      [Armor2ModPlugCategories.gauntlets]: 'LB.Gauntlets',
+      [Armor2ModPlugCategories.chest]: 'LB.Chest',
+      [Armor2ModPlugCategories.leg]: 'LB.Legs',
+      [Armor2ModPlugCategories.classitem]: 'LB.ClassItem',
+      ['seasonal']: 'LB.Seasonal'
+    };
```

Comment:
I'd suggest hoisting this out of the function, but also, you should use `t()` right here to let the i18next-scanner that you use those strings. Otherwise, it'll remove those translations automatically.

Reply:
Done